generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/AtomicConstraintConstructTuple.java

```diff
@@ -0,0 +1,33 @@
+package com.scottlogic.deg.generator.fieldspecs;
+
+import com.scottlogic.deg.generator.constraints.atomic.AtomicConstraint;
+
+import java.util.Objects;
+
+public class AtomicConstraintConstructTuple {
+    private final AtomicConstraint atomicConstraint;
+    private final boolean matchFullString;
+    private final boolean negate;
+
+    AtomicConstraintConstructTuple(AtomicConstraint atomicConstraint, boolean matchFullString, boolean negate) {
+        this.atomicConstraint = atomicConstraint;
+        this.matchFullString = matchFullString;
+        this.negate = negate;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
I would always put the contents of an if in braces

Reply:
Normally I would however this is consistent with the existing equals implementation that we have in other types throughout the project.